### ISSUE TITLE
CalcSpatialInertia(TriangleSurfaceMesh) geometry has a nonsensical default density

### ISSUE THREAD

**Issue by SeanCurtis-TRI** [RobotLocomotion/drake#21728]
### What happened?

There are two functions in `geometry_spatial_inertia.h|cc` that compute spatial inertias from geometric representation. Both take the a geometry specification (`Shape` and `TriangleSurfaceMesh`) and both take `density` value. However, for the mesh type, the density is optional and defaults to 1.

First, a density of 1 is silly. That is basically [the density of air](https://www.thoughtco.com/density-of-common-substances-2698949) which is not generally anything we expect to manipulate.

Second, the fact that one of these overloads *requires* density and the other does not is inconsistent.

Proposed solution:

When specifying arbitrary geometry and requesting spatial inertia, it is perfectly reasonable to make the user provide a density. Therefore, to remove an absurd default value and make the overloads consistent, we should deprecate the function with the default density, replacing it with a non-default density variant.

The deprecation strategy is simple.

Replace this function: 

```c++
SpatialInertia<double> CalcSpatialInertia(
    const geometry::TriangleSurfaceMesh<double>& mesh, double density = 1.0);
```

With two functions (deprecating the default density overload):
```c++
// DEPRECATE THIS ONE!!!
SpatialInertia<double> CalcSpatialInertia(
    const geometry::TriangleSurfaceMesh<double>& mesh);

SpatialInertia<double> CalcSpatialInertia(
    const geometry::TriangleSurfaceMesh<double>& mesh, double density);
```

**Comment by j4yyousi** [comment]
Hi, since 
```C++
SpatialInertia<double> CalcSpatialInertia(
    const geometry::TriangleSurfaceMesh<double>& mesh);
```
is deprecated and removed by PR #21738, should this issue be closed?

**Comment by jwnimmer-tri** [comment]
Yes, thanks.  Closing as fixed in #21738.  (The PR didn't use the correct keyword to auto-close the issue.)
